base/compiler/ssair/inlining.jl

```diff
@@ -1194,6 +1194,7 @@ function process_simple!(ir::IRCode, idx::Int, state::InliningState, todo::Vecto
             stmt.head = :invoke_modify
             pushfirst!(stmt.args, case.invoke)
             ir.stmts[idx][:inst] = stmt
```

Comment:
It might be better to include these manipulations into `optimize_modifyop!` (and some preceded condition checks as well).

Reply:
Implemented in ae696af91dede4ff1888d2fb94f08389293de7e0